sympy/functions/elementary/integers.py

```diff
@@ -132,11 +132,21 @@ def _eval_number(cls, arg):
             return arg.approximation_interval(Integer)[0]
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        arg = self.args[0]
+        arg = self.args[0].as_leading_term(x, logx=logx, cdir=cdir)
         arg0 = arg.subs(x, 0)
+        if arg0 is S.NaN:
+            arg0 = arg.limit(x, 0, dir='-' if cdir == -1 else '+')
+        r = self._eval_nseries(x, n=1, logx=None, cdir=cdir)
         if arg0.is_finite:
-            return self._eval_nseries(x, n=1, logx=None, cdir=cdir)
-        return arg.as_leading_term(x)
+            if r == arg0:
+                direction = (arg - arg0).leadterm(x, cdir=cdir)[0]
+                if direction.is_positive and cdir != -1:
```

Comment:
`cdir` cannot be used in this condition. It is the direction of `x`, not of `arg` (which may be `-x`).

I think that `dir()` could be applied here.